### PR TITLE
Upgrade react native

### PR DESCRIPTION
**Pull request by Tug**
Requires https://github.com/wordpress-mobile/react-native-recyclerview-list/pull/10
Requires https://github.com/wordpress-mobile/react-native-aztec/pull/74

This PR upgrades react-native to `v0.57.5` as well as react to `v16.6.1` (`react-native@0.57.5`'s  peer dependency version).

It does the necessary updates in the babel config, dependencies and submodules.
It also replaces the babel transform that was importing react automatically when JSX was found with a custom one importing the gutenberg package instead (see https://github.com/wordpress-mobile/gutenberg-mobile/pull/184)

### Testing Instructions
- Check that nothing breaks:
- `yarn install` (Installing the required dependencies)
- `yarn start` (Starting the metro server)
- `yarn ios` (Running the app for ios)
- `yarn android` (Running the app for android)
- `bash ./.travis/travis-checks-js.sh` (Tests)

### CODE REVIEW

**Comment by etoledom** [comment]
**As a note**
It's also important to update this:
https://github.com/wordpress-mobile/gutenberg-mobile/blob/66f244bb82766bd7787a00918ca817e5095b3388/react-native-gutenberg-bridge/RNReactNativeGutenbergBridge.podspec#L2

Thank you for taking care of this!

**Comment by gziolo** [comment]
Nice :)

File: babel.config.js [hunk 1]
```diff
@@ -21,6 +20,32 @@ module.exports = function( api ) {
 				},
 			],
 		],
+		overrides: [
```
Comment (gziolo) [inline babel.config.js]: Hope it works properly :)

File: package.json [hunk 2]
```diff
@@ -6,14 +6,20 @@
     "jsfiles": "./*.js src/*.js src/**/*.js src/**/**/*.js",
     "scssfiles": "src/*.scss src/**/*.scss"
   },
+  "engines": {
+    "node": ">=8.0.0 <9.0.0",
```
[the remark this answers is not in the document]
Comment (mzorz) [reply package.json]: oh 🤦‍♂️ I was confused and thought we were on `babel.config.js` - ok updating the README 👍

**Pull request merged**